app/inbox/views.py

```diff
@@ -0,0 +1,121 @@
+import json
+from inbox.models import Notification
+from django.contrib.staticfiles.templatetags.staticfiles import static
+from django.template.response import TemplateResponse
+from django.utils.translation import gettext_lazy as _
+from django.http import JsonResponse
+from django.core.paginator import Paginator
+from django.views.decorators.csrf import csrf_exempt
+from django.views.decorators.http import require_GET, require_http_methods
+from django.contrib.auth.decorators import login_required
+
+
+@login_required
+@require_GET
+def notifications(request):
+    """Handle all notifications."""
+    limit = int(request.GET.get('limit', 10))
+    page = int(request.GET.get('page', 1))
+    all_notifs = Notification.objects.filter(to_user_id=request.user.id).order_by('-id')
+    params = dict()
+    all_pages = Paginator(all_notifs, limit)
+    if page > 0 and page <= all_pages.num_pages:
+        all_notifications = []
+        for i in all_pages.page(page):
+            new_notif = i.to_standard_dict()
+            new_notif['username'] = i.from_user.username
+            all_notifications.append(new_notif)
+        params['data'] = all_notifications
+        params['has_next'] = all_pages.page(page).has_next()
+        params['count'] = all_pages.count
+        params['num_pages'] = all_pages.num_pages
+    else:
+        params['num_pages'] = all_pages.num_pages
+    return JsonResponse(params, status=200, safe=False)
+
+
+@login_required
+@require_http_methods(['DELETE'])
+@csrf_exempt
+def delete_notifications(request):
+    """For deleting a notification."""
+    params = {'success': []}
+    try:
+        req_body = json.loads(request.body.decode('utf-8'))
+    except:
+        pass
+    if 'delete' in req_body:
+        for i in req_body['delete']:
+            try:
+                obj = Notification.objects.get(id=i)
+                if obj.to_user.id == request.user.id:
+                    obj.delete()
```

Comment:
If we're receiving a list of notification IDs here, can we modify this to simply bulk delete? Iterating through the list is going to be slower than querying for `id__in` using the list from `req_body['delete']` and calling `.delete()` on the whole QS.

Then you could completely omit `if obj.to_user.id == request.user.id` by adding the `to_user=request.user` to the query (if they pass an ID they don't own, it won't be included in the QS that `delete` is called on.